src/ripple/protocol/STBase.h

```diff
@@ -157,7 +157,7 @@ class STBase
     template <class T>
     static
     STBase*
-    emplace(std::size_t n, void* buf, T&& val)
+    emplace(std::size_t n, void* buf, T&& val) noexcept
```

Comment:
`new(buf) U(std::forward<T>(val))` is not noexcept for `emplace<ripple::STBlob>`

Reply:
Placing  
```
static_assert(noexcept(U(std::forward<T>(val))), "");
```
and letting it fire for all the contexts that make it fail made it easy to find and fix `STBlob`for which the copy-assignment operator is not `noexcept`. It appears that it could be made fine for most other protocol types. However, `STLedgerEntry` is where things go awry: when we look up a ledger entry type and do not find it, we throw `std::runtime_error`.  So the show may stop here with `noexcept`. I could mark `noexcept` conditionally for `emplace`, `move`, and the caller of `STBase::move`.

Reply:
The heap-based `new` is going to be a problem.  If the `STBase`-based classes are going to be done, lets put that work in its own PR.

Reply:
I agree with Howard on this. The `ST*` classes are harder to get done. I'd rather we separated them out into their own PR as well.